src/components/events/EventCard.jsx

```diff
@@ -1,24 +1,54 @@
-import { Link } from "react-router-dom";
+import { useSelector } from "react-redux";
+import { selectToken } from "../auth/authSlice";
+import { useUpdateEventMutation } from "./eventsSlice";
+import { useState, useEffect } from "react";
 
-//Tooken import and selector
-// import { useSelector } from "react-redux";
-// import { selectToken } from "../../store/authSlice";
+import { Link } from "react-router-dom";
+import { useGetMeQuery } from "../users/userSlice";
 
 export default function EventCard({ event }) {
   const date = event.startTime.slice(0, 10);
   const time = event.startTime.slice(11, 19);
 
-  // const token = useSelector(selectToken);
+  const token = useSelector(selectToken);
+
+  const { data: user, isLoading, error } = useGetMeQuery();
+  console.log(user);
+
+  //state for changing button classnames when going or not going
+  const [attending, setAttending] = useState(false);
 
-  const token = true;
-  const attending = true;
+  /*Using use effect to only run the logic when the user changes or the attending users
+   Change to run the code only when its needed*/
+  useEffect(() => {
+    if (user) {
+      const isAttending = user.attendingEvents?.some(
+        (attendingEvents) => attendingEvents.id === event.id
+      );
+      setAttending(isAttending);
+    }
+  }, [user]);
+
+  const [updateEvent] = useUpdateEventMutation();
+
+  const toggleAttendance = async (newStatus) => {
+    try {
+      await updateEvent({
+        id: event.id,
+        attending: newStatus,
+      });
+      setAttending(newStatus);
+    } catch (error) {
+      console.error("Error updating attendance:", error);
+    }
+  };
 
   return (
     <li className="event-card">
-      <div className="event-details">
-        <a href={event.url}>
-          <h3>{event.name}</h3>
-        </a>
+      <div className="event-details-div">
+        <Link to={`/events/${event.id}`}>
+          <h3 className="event-name">{event.name}</h3>
+        </Link>
         <p>
           <b>Date:</b> {date}
```

Comment:
I read somewhere that to bold text, it is more conventional to use css styling - Ex `font-weight: 700;`. I am not 100% on that though

Reply:
We can come back to change that if we decide to do it sure. But I did it like this for now, cause the only way I thought on doing this with css for just two words was to put them inside any <tag/> and style that tag with className, but the <b> tags its doing that already.

Reply:
we can style them separately for more personalization or use that  - Ex font-weight: 700;